discretize/operators/differential_operators.py

```diff
@@ -1112,83 +1207,141 @@ def _aveN2Ey(self):
             if self.dim == 1:
                 self._aveN2E = self._aveN2Ex
             elif self.dim == 2:
-                self._aveN2E = sp.vstack((
-                    self._aveN2Ex, self._aveN2Ey
-                ), format="csr")
+                self._aveN2E = sp.vstack((self._aveN2Ex, self._aveN2Ey), format="csr")
             elif self.dim == 3:
-                self._aveN2E = sp.vstack((
-                    self._aveN2Ex, self._aveN2Ey, self._aveN2Ez
-                ), format="csr")
+                self._aveN2E = sp.vstack(
+                    (self._aveN2Ex, self._aveN2Ey, self._aveN2Ez), format="csr"
+                )
         return self._aveN2E
 
     @property
     def _aveN2Fx(self):
         if self.dim == 1:
-            aveN2Fx = av(self.nCx)
+            aveN2Fx = av(self.shape_cells[0])
         elif self.dim == 2:
-            aveN2Fx = sp.kron(av(self.nCy), speye(self.nNx))
+            aveN2Fx = sp.kron(av(self.shape_cells[1]), speye(self.shape_nodes[0]))
         elif self.dim == 3:
-            aveN2Fx = kron3(av(self.nCz), av(self.nCy), speye(self.nNx))
+            aveN2Fx = kron3(
+                av(self.shape_cells[2]),
+                av(self.shape_cells[1]),
+                speye(self.shape_nodes[0]),
+            )
         return aveN2Fx
 
     @property
     def _aveN2Fy(self):
         if self.dim == 1:
             return None
         elif self.dim == 2:
-            aveN2Fy = sp.kron(speye(self.nNy), av(self.nCx))
+            aveN2Fy = sp.kron(speye(self.shape_nodes[1]), av(self.shape_cells[0]))
         elif self.dim == 3:
-            aveN2Fy = kron3(av(self.nCz), speye(self.nNy), av(self.nCx))
+            aveN2Fy = kron3(
+                av(self.shape_cells[2]),
+                speye(self.shape_nodes[1]),
+                av(self.shape_cells[0]),
+            )
         return aveN2Fy
 
     @property
     def _aveN2Fz(self):
         if self.dim == 1 or self.dim == 2:
             return None
         else:
-            aveN2Fz = kron3(speye(self.nNz), av(self.nCy), av(self.nCx))
+            aveN2Fz = kron3(
+                speye(self.shape_nodes[2]),
+                av(self.shape_cells[1]),
+                av(self.shape_cells[0]),
+            )
         return aveN2Fz
 
     @property
-    def aveN2F(self):
+    def average_node_to_face(self):
         """
         Construct the averaging operator on cell nodes to cell faces, keeping
         each dimension separate.
         """
-        if getattr(self, '_aveN2F', None) is None:
+        if getattr(self, "_aveN2F", None) is None:
             # The number of cell centers in each direction
             if self.dim == 1:
                 self._aveN2F = self._aveN2Fx
             elif self.dim == 2:
-                self._aveN2F = sp.vstack((
-                    self._aveN2Fx, self._aveN2Fy
-                ), format="csr")
+                self._aveN2F = sp.vstack((self._aveN2Fx, self._aveN2Fy), format="csr")
```

Comment:
similar comment as earlier about possibly renaming the private variables to be the same as the public ones